lib/api_watch_diff.ml

```diff
@@ -11,7 +11,11 @@ and item_change =
   | Value of { name : string; change : value_description change }
   | Module of module_diff
 
-and module_change = Unsupported | Supported of item_change list
+and module_change =
+  | Unsupported
+  | Supported of item_change list
+  | Mod_added of module_declaration
+  | Mod_removed of module_declaration
```

Comment:
How about instead reworking the `'item change` type. I think it's about the right time to do this.

We could instead define the following types:
```ocaml
type ('item, 'diff) diff =
  | Added of 'item
  | Removed of 'item
  | Modified of 'diff
  
 (** The simplest diff representation for the modification of a value of type 'a.
     [reference] is the value before and [current] is the value after the change occured.
     Use this type when there is no better representation available. *)
 type 'a atomic_modification =
   { reference : 'a
    ; current : 'a
    }
  
  type value_diff =
    { vname : string
    ; vdiff : (value_description, value_description atomic_modification) diff
    }
  
  type module_diff =
    { mname : string
    ; mdiff : (module_declaration, module_modification) diff
    }
  and module_modification =
    | Unsupported
    | Supported of item_diff list 
  and item_diff =
    | Value of value_diff
    | Module of module_diff
 ```
 
 This uses more consistent naming and structure and allows for better representation. What do you think?

Reply:
Yes, this definitely has better structure than the current implementation! It can also be used better when we add support for more `signature_items`.